clone/internal/download/batch.go

```diff
@@ -48,25 +47,34 @@ func Bulk(ctx context.Context, first, treeSize uint64, batchFetch BatchFetch, wo
 	rangeChans := make([]chan workerResult, workers)
 
 	increment := workers * batchSize
+
+	waitCh := make(chan struct{})
```

Comment:
Instead of creating this channel and then doing alignment inside the concurrency below, I imagined that this change would do the alignment in the main thread here, and once it had got alignment, would then kick off the concurrency by continuing on with an updated `start`. Is this possible?

I have to jump to meetings now, but LMK if this doesn't make sense.